### ISSUE TITLE
Database search fails to find any hits 8.0.21

### ISSUE THREAD

**Issue by eugenemel** [eugenemel/maven#34]
@PMSeitzer  It looks like library search is not pickup any hits.  here is trace 

```
peakdetectiondialog::runBackgroundJob(QString funcName) Completed.
BackgroundPeakUpdate:processCompounds(vector<Compound*> set, string setName)
The PeakGroupCompoundMatchingPolicy is set to ALL_MATCHES
BackgroundPeakUpdate::processCompoundSlices(vector<mzSlice*>&slices, string setName)
Processing slices: setName= compounds  slices= 161
processCompoundSlices() Slices= 161
processCompoundSlices() EICs= 1610
processCompoundSlices() Groups= 0
processCompoundSlices() Peaks= 0
Compound <--> Peak Group Matches= 0
BackgroundPeakUpdate:processCompoundSlices() done.  1518  sec.
```

**Comment by PMSeitzer** [comment]
Hey Eugene, thanks for reporting this.

Can you provide a few additional pieces of information, so I can reproduce this on my machine?

Specifically,
(1) What files were run to produce the bug
and
(2) What parameters were used in the analysis?

**Comment by PMSeitzer** [comment]
I tested this out on my machine, and it seems to be working properly.

Is it possible that the library being searched against does not contain compounds with molecular formulas?

**Comment by PMSeitzer** [comment]
The groups could be getting filtered out. I will add some `debug()` messages to help determine if that is the problem.

**Comment by PMSeitzer** [comment]
It looks like the problem is that all of the groups are being filtered out.

**Comment by PMSeitzer** [comment]
The library searched did not contain any entry for the `precursorMz` field, so the value remained at its default value of `0`.  The `precursorMz` is used to convert the ppm mass tolerance into Da, and was resulting in a maximum fragment tolerance in `Da` of `0 Da`, so no fragments were ever being matched.

I have updated the code so that if the `precursorMz` is missing from the library, the `precursorMz` of the MS/MS scan should be used to determine fragment tolerance in Da.  I have tested this and it seems to be working properly.  Released as `8.0.22`.
